### PR TITLE
chore: Move bloomstack core utils.py function to erpnext apps

### PR DESCRIPTION
**Pull request by vishdha**
TASK ID: https://bloomstack.com/desk#Form/Task/TASK-2021-00282
Moved bloomstac_core utils.py functions to erpnext apps.
this is an independent file.
Dependency: No
gif required: No.
whoever has reference to bloomstack_core.utils function kindly refer this pr

Bloomstack_core PR: https://github.com/Bloomstack/bloomstack_core/pull/698

### CODE REVIEW

File: erpnext/utils.py [hunk 1]
```diff
@@ -52,4 +53,75 @@ def create_authorization_request(dt, dn, contact_email, contact_name):
 	new_authorization_request.linked_docname = dn
 	new_authorization_request.authorizer_email = contact_email
 	new_authorization_request.authorizer_name = contact_name
-	new_authorization_request.save()
\ No newline at end of file
+	new_authorization_request.save()
+
+@frappe.whitelist(allow_guest=True)
```
Comment (sahil28297) [inline erpnext/utils.py]: why does this function has `allow_guest=True`, the login_as button is added on the user doctype and the user is already logged in at this point.

File: erpnext/utils.py [hunk 2]
```diff
@@ -52,4 +53,75 @@ def create_authorization_request(dt, dn, contact_email, contact_name):
...
+		frappe.local.login_manager.login_as(user)
+		frappe.set_user(user)
+
+		frappe.db.commit()
+		frappe.local.response["redirect_to"] = '/'
+		return True
+
+	return False
+
+def get_abbr(txt, max_length=2):
```
Comment (sahil28297) [inline erpnext/utils.py]: There's an exact same function which already exists in erpnext `erpnext/erpnext/utilities/utils.py` line 3

File: erpnext/setup/utils.py [hunk 3]
```diff
@@ -141,6 +141,59 @@ def insert_record(records):
...
+
+		email_authorized_doc(docname)
+
+def email_authorized_doc(authorization_request_name):
+	authorization_request = frappe.get_doc("Authorization Request", authorization_request_name)
+	authorized_doc = frappe.get_doc(authorization_request.linked_doctype, authorization_request.linked_docname)
+	recipients = [authorization_request.authorizer_email]
+	company = authorized_doc.company if hasattr(authorized_doc, 'company') else get_default_company()
+	subject = "Your signed {0} with {1}".format(authorized_doc.doctype, company)
+	message = frappe.render_template("templates/emails/authorization_request.html", {
```
Comment (sahil28297) [inline erpnext/setup/utils.py:185]: The path to the html file is wrong, you'll have to give the full path

**Review by sahil28297 — CHANGES_REQUESTED** [review]

File: erpnext/setup/utils.py (hunk 3 above)
Comment (vishdha) [reply erpnext/setup/utils.py:185]: This is the correct path. autorization_request.html needs to move in the emails folder. whoever is moving authoirzation_request.html need to use this as dependent PR

File: erpnext/utils.py (hunk 2 above)
Comment (vishdha) [reply erpnext/utils.py]: Yeah. We have to remove one of them. this is used for Item.py which again contains autoname and custom_autoname function but the code is the same. 
so one of the get_abbr functions we need to remove.

Whoever moving item.py in bloomstack_core will be used as dependent PR

File: erpnext/utils.py (hunk 1 above)
Comment (vishdha) [reply erpnext/utils.py]: @sahil28297 I don't know the use case why they enabled allow_guest=True I will check and update in PR

File: erpnext/utils.py (hunk 2 above)
Comment (sahil28297) [reply erpnext/utils.py]: You can remove the function from this PR, the one moving item.py will use the original erpnext function

File: erpnext/setup/utils.py (hunk 3 above)
Comment (sahil28297) [reply erpnext/setup/utils.py:185]: the relative path is wrong, there's no templates folder inside `erpnext/setup` folder the relative path is `../templates/emails/authorization_request.html`, that is why I asked you to put the full path

File: erpnext/utils.py (hunk 1 above)
Comment (sahil28297) [reply erpnext/utils.py]: any update on this?

Comment (vishdha) [reply erpnext/utils.py]: hey @sahil28297  i messaged @DeveloperMetal  regarding this if i get reply will update or simply i will remove this allow_guest by evening

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**